tests/Unit/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProviderTest.php

```diff
@@ -101,11 +108,61 @@ public function testGetData(array $productData, array $expectedData)
         $this->assertEquals($expectedData, $formData);
     }
 
-    public function getExpectedData()
+    public function getExpectedData(): Generator
```

Comment:
@zuk3975 you nailed this part! I love how you improved the provider functions I really have some progress to make regarding the use of `yield` ^^

I tried doing something similar some time ago (mixing loop and yield) but without success, it's quite interesting to see it in action!